pinot-spi/src/main/java/org/apache/pinot/spi/trace/Tracing.java

```diff
@@ -267,5 +279,20 @@ public static void initializeThreadAccountant(int numPqr, int numPqw, PinotConfi
     public static boolean isInterrupted() {
       return Thread.interrupted() || Tracing.getThreadAccountant().isAnchorThreadInterrupted();
     }
+
+    public static void sampleAndCheckInterruption() {
+      if (isInterrupted()) {
+        throw new EarlyTerminationException("Interrupted while merging records");
```

Comment:
Is the exception message clear enough regardless of whether this happened on broker or server ? I am guessing call stack will make it apparent

Reply:
The one captures this interruption will throw/return with more detailed information. It's hard for us to derive the execution context here.